### ISSUE TITLE
Integration with Sequelize

### ISSUE THREAD

**Issue by mark-lester** [PaulUithol/Backbone-relational#542]
Being able to run the same data definition code on both sides is the coup de gras for Node imo.   

It's easer to generate Sequelize constructors from the Backbone-Relational defs rather than the other way about. But that's a little back to front. The data source is the DB, not the client.
The are also extra things like UI attributes e.g. should display, and validation stuff like "must be greater than such a  field", e.g. the start time has to be less than or equal to the end, which I want to pass on to something like handlebars-formhandler (with extensions if required) .
I propose to use a 3rd file, a super def, and generate BBR and Sequelize defs from that.  Unless of course you can de-duplicate me.

**Comment by mark-lester** [comment]
well that didnt go down too well.
I have enhanced sequelize-restful so that it transmits the relational information. (the HEAD stuff didnt work for a number of reasons). I can now build a set of Model and Collection definitions based on that data (i.e I have a collection of Model definitions)
I am now trying to work out how one specifies foreign keys in backbone-relational. I dont have nested data from my interface, I've got /api/<model>/[id][/<submodel>/[sub id]] . which return rows of model, or submodel if requested. So now I want to get to the parent models data. It's not that hard, I of course have a solution, but I'd really like to unlock a shed load of lazy fetching and other cool stuff.
If I've got the following models (psuedo code)
Agency = {id:,name:}
Route = {id:AgencyId:,...}
how do I set stuff up so I can go something along the lines of
 route.get("Agency").get("name")


or better  route.get("AgencyName")
